includes/REST_API/Link_Controller.php

```diff
@@ -161,6 +161,21 @@ public function parse_link( $request ) {
 			'description' => '',
 		];
 
+		// Do not request instagram.com, as it redirects to a login page.
+		// See https://github.com/GoogleForCreators/web-stories-wp/issues/10451.
+		$matches = [];
+		if ( preg_match( '~^https?://(www\.)?instagram\.com/([^/]+)/?$~', $url, $matches ) ) {
```

Comment:
I see two simple solutions to account for the scenario Felipe has mentioned:

**Option A:** Remove query params before doing the regex.

```suggestion
		$url_without_query = explode( '?', $url )[0];
		if ( preg_match( '~^https?://(www\.)?instagram\.com/([^/]+)/?$~', $url_without_query, $matches ) ) {
```

(Yes, `$matches = []` can be safely removed! My bad for suggesting that initially. There is no need to define that variable beforehand!)

**Option B:** Update regex to account for query param.

```suggestion
		$matches = [];
		if ( preg_match( '~^https?://(www\.)?instagram\.com/([^/]+)/?(\?.*)?$~', $url, $matches ) ) {
```

Reply:
I tend towards option A because it allows us to keep the regex simple and understandable.